### DISCUSSION TITLE
[BUG] access denied during get-pnpweb

### DISCUSSION THREAD

**Discussion by mikarin20** [pnp/powershell#4226] [General]
### Reporting an Issue or Missing Feature
Get-pnpweb throw access denied error

Access is denied. (Exception from HRESULT: 0x80070005 (E_ACCESSDENIED))

### Expected behavior 
it was running before normally before the latest change regarding app register procedure. i able to see that connection established normally because get-pnplist codes working fine

get-pnpweb works but
 $web = Get-PnPWeb -Includes Webs, Title, Url throws and error

### Actual behavior 
Access is denied. (Exception from HRESULT: 0x80070005 (E_ACCESSDENIED))

(https://github.com/user-attachments/assets/99154b4d-6f36-4109-a92a-c2b217f3d3ed)

### Steps to reproduce behavior
<script src="https://gist.github.com/mikarin20/773517c68f39149308f318122ef00acf.js"></script>

### What is the version of the Cmdlet module you are running?
(you can retrieve this by executing ```Get-Module -Name "PnP.PowerShell" -ListAvailable```)
2.10.0
### Which operating system/environment are you running PnP PowerShell on?
- [x ] Windows
- [ ] Linux
- [ ] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by gautamdsheth** [comment]
This means that you are missing some permissions in your EntraID app. How are you connecting to SharePoint ? Using `Connect-PnPOnline` ?

**Reply by mikarin20** [reply]
here my connection strings and permissions.


$SiteURL = "*"
$tenantID= "*"
$clientID= "*"
#$thumbprint= "*"
#$LibraryName = "Preservation Hold Library"
$certificatepath = "\1-List Hold Libraries\PnpForSharepoint.pfx"

#Connect-PnPOnline -url $siteurl
Connect-PnPOnline -ClientId $clientID  -url $siteUrl -tenant $tenantID -CertificatePath $certificatepath


![image](https://github.com/user-attachments/assets/56c4c447-392f-479d-a79f-bd490e602e4f)
